Filogic: arm-trusted-firmware-tools: compilation broken on Alpine since update to version 2.9

### Describe the bug

Since the upgrade of the arm-trusted-firmware-tools to version 2.9, the compilation for Filogic target fails on Apline Linux. The relevant error message is as follows:

```
  HOSTCC  fiptool.c
fiptool.c: In function 'parse_fip':
fiptool.c:320:39: error: overflow in conversion from 'long unsigned int' to 'int' changes value from '2148012658' to '-2146954638' [-Werror=overflow]
  320 |                 if (ioctl(fileno(fp), BLKGETSIZE64, &st_size) == -1)
      |                                       ^~~~~~~~~~~~
cc1: all warnings being treated as errors
make[4]: *** [Makefile:82: fiptool.o] Error 1
```

I only tested using the 23.05 branch, but I suspect the same happens with Master.
FWIW, I'm using Alpine 3.18.3 in WSL2.
Compilation using other recent distros (e.g. Debian Bookworm) still works fine.

### OpenWrt version

23.05.0-rc3

### OpenWrt target/subtarget

Mediatek / Filogic

### Device

Xiaomi AX6000

### Image kind

Self-built image

### Steps to reproduce

Compile OpenWrt 23.05.0-rc3, 23.05-SNAPSHOT or any revision starting with 6ad85a6c2d733e9b06e0e2b932428cf0d171ff56 for Mediatek Filogic target using the latest Alpine Linux as build host.

### Actual behaviour

Compilation fails with the error message noted above.

### Expected behaviour

Compilation succeeds using any supported distro, including latest Alpine Linux.

### Additional info

n/a

### Diffconfig

```text
n/a
```


### Terms

- [X] I am reporting an issue for OpenWrt, not an unsupported fork.

--- a/package/boot/arm-trusted-firmware-tools/Makefile
+++ b/package/boot/arm-trusted-firmware-tools/Makefile
@@ -29,10 +29,16 @@ define Package/arm-trusted-firmware-tools
   BUILDONLY:=1
 endef
 
+ifeq ($(shell uname -s),Linux)
+	HOST_MUSL_FLAG:=$(shell [ "$$(2>&1 ldd --version | head -n 1 | cut -d ' ' -f1)" = "musl" ] && echo " -D__MUSL__")
+else
+	HOST_MUSL_FLAG:=
+endif
+
 define Host/Compile
 	$(MAKE) -C \
 		$(HOST_BUILD_DIR)/tools/fiptool \
-		CPPFLAGS="$(HOST_CFLAGS)" \
+		CPPFLAGS="$(HOST_CFLAGS)$(HOST_MUSL_FLAG)" \
 		LDFLAGS="$(HOST_LDFLAGS)" \
 		OPENSSL_DIR="$(STAGING_DIR_HOST)"
 endef

--- a/package/boot/arm-trusted-firmware-tools/patches/003-fiptool_musl_fix.patch
+++ b/package/boot/arm-trusted-firmware-tools/patches/003-fiptool_musl_fix.patch
@@ -1,0 +1,14 @@
+--- a/tools/fiptool/fiptool.c
++++ b/tools/fiptool/fiptool.c
+@@ -318,7 +318,11 @@
+ 
+ #ifdef BLKGETSIZE64
+ 	if ((st.st_mode & S_IFBLK) != 0)
++#ifdef __MUSL__
++		if (ioctl(fileno(fp), (int)BLKGETSIZE64, &st_size) == -1)
++#else
+ 		if (ioctl(fileno(fp), BLKGETSIZE64, &st_size) == -1)
++#endif
+ 			log_err("ioctl %s", filename);
+ #endif
+ 

--- a/package/network/utils/nftables/patches/001-for-bash-syntax.patch
+++ b/package/network/utils/nftables/patches/001-for-bash-syntax.patch
@@ -1,0 +1,11 @@
+--- a/configure
++++ b/configure
+@@ -17940,7 +17940,7 @@
+ 	echo "	${STABLE_RELEASE}"
+ 	echo "};"
+ 	echo "static char nftbuildstamp[] = {"
+-	for ((i = 56; i >= 0; i-= 8)); do
++	for i in `seq 56 -8 0`; do
+ 		echo "	((uint64_t)MAKE_STAMP >> $i) & 0xff,"
+ 	done
+ 	echo "};"

--- a/tools/bc/patches/001-getopt-declaration.patch
+++ b/tools/bc/patches/001-getopt-declaration.patch
@@ -1,0 +1,11 @@
+--- a/lib/getopt.c
++++ b/lib/getopt.c
+@@ -197,7 +197,7 @@ static char *posixly_correct;
+    whose names are inconsistent.  */
+ 
+ #ifndef getenv
+-extern char *getenv ();
++extern char *getenv (const char *);
+ #endif
+ 
+ #endif /* not __GNU_LIBRARY__ */

